tests/common/include/tests/common/tools/particle_gun.hpp

```diff
@@ -36,10 +36,13 @@ struct particle_gun {
     ///         intersections of the surfaces that were encountered.
     template <typename detector_t, typename trajectory_t>
     DETRAY_HOST_DEVICE inline static auto shoot_particle(
-        const detector_t &detector, const trajectory_t &traj) {
+        const detector_t &detector, const trajectory_t &traj,
+        typename detector_t::scalar_type mask_tolerance =
```

Comment:
I would store `mask_tolerance` as a private member or make an internal option struct, 

```c++
particle_gun gun;
gun.set_mask_tolerance(15 * um);
```

Reply:
Right now this function is static and there is no particle gun object created, but I can package the tolerance it in a config struct